app.ts

```diff
@@ -0,0 +1,187 @@
+//#region Global Variables
+let resizeTimer = 50;
+let previousCursor: number[];
+let previousScale: number;
+
+//#endregion Global Variables
+
+//#region AppLogic
+//#region API
+async function getRecordCount() : Promise<number> {
+    const response = await fetch('http://localhost:2050/recordCount');
+    return await response.json();
+}
+
+async function getColumnNames() : Promise<string[]>{
+    const response = await fetch('http://localhost:2050/columns');
+    return await response.json();
+}
+
+async function getRecords(fromID: number, toID: number): Promise<string[][]> {
+    const response = await fetch(`http://localhost:2050/records?from=${(fromID).toString()}&to=${(toID).toString()}`);
+    return await response.json();
+}
+
+//#endregion API
+
+//#region Data Loading methods
+
+async function placeRecords(fromID: number, toID: number): Promise<number[]> {
+    const records = await getRecords(fromID, toID)
+    let appendable = '';
+    for (const record of records) {
+            appendable += `<tr id="table-row-${record[0]}">`;
+        for (const column of record) {
+            appendable += `<td align="center">${column.toString()}</td>`;     
+        }
+        appendable += '</tr>';
+    }
+    $("#WrapperTable-ContentBody").empty();
+    $("#WrapperTable-ContentBody").append(appendable);
+    return await [fromID as number, toID as number];
+}
+
+// Credit: https://gist.github.com/ca0v/73a31f57b397606c9813472f7493a940
+
+async function placeRecordsFromCursor(cursor: number[]): Promise<number[]> {
+    cursor = cursor.sort((a,b) => {return a-b});
+    return await placeRecords(cursor[0], cursor[1]);
+}
+//#endregion Data Loading methods
+
+//#region Handlers
+async function getPageContent(fromID: number, toID: number): Promise<number[]> {
+    $("#WrapperTable-HeaderRow").empty();
+    getColumnNames()
+        .then((columns: string[]) => {
+            for (const column of columns) {
+                let writable: string = `<th align="center">${column}</th>`;
+                $("#WrapperTable-HeaderRow").append(writable);
+            }
+        });
+    return await placeRecords(fromID, toID);
+}
+
+function toNumber(input: string | number, output: any = 1) : number {
```

Comment:
I find the `output` parameter a bit weird. It looks like a flag to either parse the value as an integer or a float. But then a better way to do this would be to name it `parseAsFloat` and declare it as a `boolean`.

The only time I see the flag used for parsing a float was for user-input. I don't see a user typing in a floating point number as an index; would actually expect an error message if that was the case?

Reply:
At some point, Fritz mentioned about inserting sanity checks to make sure that the user does not insert any unwanted values into the field that could break the API call.

I imagined a scenario in which he would've tested to make sure that someone does not insert this breaking change - but I can remove it just as readily.

Reply:
Jip, agree you need to show an error message when the user types in something that doesn't make sense. But I would lean towards not having the `output` parameter at all and ignoring the decimal bit and just parsing the number as an int.